src/python/pants/bin/remote_pants_runner.py

```diff
@@ -42,7 +43,12 @@ def _forward_signal_with_timeout(self, signum, signame):
     self._pailgun_client.maybe_send_signal(signum)
 
   def handle_sigint(self, signum, _frame):
-    self._forward_signal_with_timeout(signum, 'SIGINT')
+    if self._pailgun_client._maybe_last_pid():
+      self._forward_signal_with_timeout(signum, 'SIGINT')
+    else:
+      # NB: We consider not having received a PID yet as "not having started substantial work".
+      # So in this case, we let the client die gracefully, and the server handle the closed socket.
+      super(PailgunClientSignalHandler, self).handle_sigint(signum, _frame)
```

Comment:
I'm not sure why this works to fix #7920, because I think thinking that if the client hasn't received the pantsd pid yet, it wouldn't know which pid to kill, so `self._forward_signal_with_timeout()` wouldn't be crashing the daemon anyway -- am I mistaken? Is it possible to add a test case?

Reply:
This case is the case where we haven't received the pid yet, so we just crash the client (via the usual handling sigint mechanism) in this case and let the closing socket crash the request on the server side.

Reply:
Oh, yes!